doc/man/man3/crypto_x25519_dangerous_fast.3monocypher

```diff
@@ -0,0 +1,135 @@
+.\" This file is dual-licensed.  Choose whichever you want.
+.\"
+.\" The first licence is a regular 2-clause BSD licence.  The second licence
+.\" is the CC-0 from Creative Commons. It is intended to release Monocypher
+.\" to the public domain.  The BSD licence serves as a fallback option.
+.\"
+.\" SPDX-License-Identifier: BSD-2-Clause OR CC0-1.0
+.\"
+.\" ----------------------------------------------------------------------------
+.\"
+.\" Copyright (c) 2020 Fabio Scotoni
+.\" All rights reserved.
+.\"
+.\"
+.\" Redistribution and use in source and binary forms, with or without
+.\" modification, are permitted provided that the following conditions are
+.\" met:
+.\"
+.\" 1. Redistributions of source code must retain the above copyright
+.\"    notice, this list of conditions and the following disclaimer.
+.\"
+.\" 2. Redistributions in binary form must reproduce the above copyright
+.\"    notice, this list of conditions and the following disclaimer in the
+.\"    documentation and/or other materials provided with the
+.\"    distribution.
+.\"
+.\" THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+.\" "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+.\" LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+.\" A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+.\" HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+.\" SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+.\" LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+.\" DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+.\" THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+.\" (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+.\" OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+.\"
+.\" ----------------------------------------------------------------------------
+.\"
+.\" Written in 2020 by Fabio Scotoni
+.\"
+.\" To the extent possible under law, the author(s) have dedicated all copyright
+.\" and related neighboring rights to this software to the public domain
+.\" worldwide.  This software is distributed without any warranty.
+.\"
+.\" You should have received a copy of the CC0 Public Domain Dedication along
+.\" with this software.  If not, see
+.\" <https://creativecommons.org/publicdomain/zero/1.0/>
+.\"
+.Dd March 24, 2020
+.Dt CRYPTO_X25519_DANGEROUS_FAST 3monocypher
+.Os
+.Sh NAME
+.Nm crypto_x25519_dangerous_fast ,
+.Nm crypto_x25519_dangerous_small
+.Nd generation of Curve25519 points including low-order points
```

Comment:
"_With_ a low order _component_".

We can't generate low order points with those because of clamping. The price for that is a slight, undetectable statistical bias. in the resulting key distribution (once you flip the sign at random).

Perhaps I wasn't clear, but the only danger with those functions is revealing 3 bits of the private scalar, and that doesn't give much leverage to anyone. (Hence my initial name, "dirty".

Did you misunderstand that bit, and maintain that we should spell those "dangerous?"

Reply:
I'm reasonably sure that did not misunderstand that bit, but instead phrased it badly.

I do maintain that the names should include “dangerous” because it's a very loud alert that people are messing with things that are, in fact, probably not where they're intending to be, but could settle for “dirty” if accuracy matters more than the warning signal.

Reply:
OK, I see. I believe I prefer "dirty", for a couple reasons:

- Your manual page already makes crystal clear how specialised and delicate those functions are. This is warning enough.
- "Dirty" is shorter than "dangerous" by 4 characters.
- The "danger" is minimal: worst case we are leaking 3 bits, that will never be used for anything else. Thus, "dirty" is more accurate.
- "Dangerous" can raise eyebrows, and get people to wonder why I considered it okay to put anything "dangerous" in a supposedly easy to use crypto library.

That last consideration is perhaps the most important one. I don't want a short sighted laser focus on those two function names to undermine the credibility of Monocypher. Something like:

    egrep "danger|warn|hack|TODO"

Reply:
Alright, okay, "dirty" it is after all.